airbyte-webapp/src/components/connectorBuilder/manifestToBuilderForm.ts

```diff
@@ -0,0 +1,384 @@
+import isEqual from "lodash/isEqual";
+
+import { AirbyteJSONSchema } from "core/jsonSchema/types";
+import {
+  CartesianProductStreamSlicer,
+  ConnectorManifest,
+  DatetimeStreamSlicer,
+  DeclarativeStream,
+  DeclarativeStreamSchemaLoader,
+  DpathExtractor,
+  HttpRequester,
+  HttpRequesterAuthenticator,
+  InlineSchemaLoader,
+  InterpolatedRequestOptionsProvider,
+  ListStreamSlicer,
+  SimpleRetriever,
+  SimpleRetrieverPaginator,
+  SimpleRetrieverStreamSlicer,
+  Spec,
+  SubstreamSlicer,
+} from "core/request/ConnectorManifest";
+
+import {
+  authTypeToKeyToInferredInput,
+  BuilderFormAuthenticator,
+  BuilderFormValues,
+  BuilderPaginator,
+  BuilderStream,
+  BuilderSubstreamSlicer,
+  DEFAULT_BUILDER_FORM_VALUES,
+  DEFAULT_BUILDER_STREAM_VALUES,
+  isInterpolatedConfigKey,
+} from "./types";
+import { formatJson } from "./utils";
+
+export const convertToBuilderFormValues = (
+  manifest: ConnectorManifest,
+  currentBuilderFormValues: BuilderFormValues,
+  streamListErrorMessage: string | undefined
+) => {
+  // TODO: replace these checks with a call to the soon-to-be /manifest/resolve endpoint, to resolve refs, options, and validate the manifest against the schema
+  if (streamListErrorMessage) {
+    let errorMessage = streamListErrorMessage;
+    if (errorMessage[0] === '"') {
+      errorMessage = errorMessage.substring(1, errorMessage.length);
+    }
+    if (errorMessage.slice(-1) === '"') {
+      errorMessage = errorMessage.substring(0, errorMessage.length - 1);
+    }
+    const tracebackIndex = errorMessage.indexOf(" - Traceback");
+    if (tracebackIndex >= 0) {
+      errorMessage = errorMessage.substring(0, tracebackIndex);
+    }
```

Comment:
This can be removed once https://github.com/airbytehq/airbyte/pull/21443 is merged

Reply:
Looks like this is merged in